.github/workflows/gradle.yml

```diff
@@ -15,21 +15,43 @@ jobs:
       - name: Check images exist
         run: ./tools/bin/check_images_exist.sh
 
-      - name: Cache java deps
+      - name: Docker Caching
         uses: actions/cache@v2
         with:
-          path: ~/.gradle
-          key: gradle-${{ hashFiles('**/*.gradle') }}
+          path: |
+            /tmp/docker-registry
+          key: docker-${{ runner.os }}-${{ hashFiles('Dockerfile') }}-${{ github.sha }}
           restore-keys: |
-            gradle-
+            docker-${{ runner.os }}-${{ hashFiles('Dockerfile') }}-
+            docker-${{ runner.os }}-
 
-      - name: Cache node deps
+      - name: Pip Caching
         uses: actions/cache@v2
         with:
-          path: ~/.npm
-          key: node-${{ hashFiles('**/package-lock.json') }}
+          path: |
+            ~/.cache/pip
```

Comment:
Depending on how you use pip, more might be stored here. Right now this is super lightweight.  It also allows restore-keys because this can be used by all venvs.